src/main/java/no/nav/fo/veilarbdialog/service/DialogDataService.java

```diff
@@ -1,85 +1,78 @@
 package no.nav.fo.veilarbdialog.service;
 
-import no.nav.apiapp.feil.UlovligHandling;
-import no.nav.apiapp.security.PepClient;
-import no.nav.dialogarena.aktor.AktorService;
-import no.nav.fo.veilarbdialog.client.KvpClient;
+import lombok.RequiredArgsConstructor;
+import no.nav.common.client.aktorregister.AktorregisterClient;
+import no.nav.common.featuretoggle.UnleashService;
+import no.nav.common.types.feil.IngenTilgang;
+import no.nav.common.types.feil.UlovligHandling;
+import no.nav.fo.veilarbdialog.auth.AuthService;
 import no.nav.fo.veilarbdialog.db.dao.DataVarehusDAO;
 import no.nav.fo.veilarbdialog.db.dao.DialogDAO;
 import no.nav.fo.veilarbdialog.db.dao.DialogFeedDAO;
 import no.nav.fo.veilarbdialog.domain.*;
-import no.nav.fo.veilarbdialog.domain.KafkaDialogMelding;
-import no.nav.sbl.featuretoggle.unleash.UnleashService;
-import org.springframework.stereotype.Component;
+import no.nav.fo.veilarbdialog.feed.KvpService;
+import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import java.util.Collections;
 import java.util.Date;
 import java.util.List;
 import java.util.Optional;
 
-@Component
+@Service
 @Transactional
-public class AppService {
+@RequiredArgsConstructor
+public class DialogDataService {
 
-    private final AktorService aktorService;
+    private final AktorregisterClient aktorregisterClient;
     private final DialogDAO dialogDAO;
     private final DialogStatusService dialogStatusService;
     private final DataVarehusDAO dataVarehusDAO;
     private final DialogFeedDAO dialogFeedDAO;
-    private final PepClient pepClient;
-    private final KvpClient kvpClient;
+    private final KvpService kvpService;
     private final UnleashService unleashService;
     private final KafkaDialogService kafkaDialogService;
-
-    public AppService(AktorService aktorService,
-                      DialogDAO dialogDAO,
-                      DialogStatusService dialogStatusService,
-                      DataVarehusDAO dataVarehusDAO,
-                      DialogFeedDAO dialogFeedDAO,
-                      PepClient pepClient,
-                      KafkaDialogService kafkaDialogService,
-                      KvpClient kvpClient,
-                      UnleashService unleashService) {
-        this.aktorService = aktorService;
-        this.dialogDAO = dialogDAO;
-        this.dialogStatusService = dialogStatusService;
-        this.dataVarehusDAO = dataVarehusDAO;
-        this.dialogFeedDAO = dialogFeedDAO;
-        this.pepClient = pepClient;
-        this.kvpClient = kvpClient;
-        this.unleashService = unleashService;
-        this.kafkaDialogService = kafkaDialogService;
+    private final AuthService auth;
+
+    private String assertAccessToAktorId(String aktorId)
+            throws IngenTilgang {
+        if (!auth.harTilgangTilPerson(aktorId)) {
+            throw new IngenTilgang(String.format(
+                    "%s har ikke lesetilgang til %s",
```

Comment:
Her også kastes det exception, men fungerer det mtp hvordan det håndteres i frontend? Veiledere skal for eksempel ikke ha tilgang til informasjon på alle brukere, men det skal ikke vises en feilmelding.

Reply:
Tanken er at det skal funke som før, da `PepClient` kastet en `IngenTilgang`.

Gleden ved unchecked exceptions.